py4j-python/src/py4j/clientserver.py

```diff
@@ -9,7 +9,8 @@
 
 from __future__ import unicode_literals, absolute_import
 
-from collections import deque, Callable
+from collections import deque
+from collections.abc import Callable
```

Comment:
The problem was that it breaks Python 2 so I used the deprecated one. Probably we can just remove the usage of `Callable`, and check if it is a weakref reference.

Reply:
That makes perfect sense! Sorry I missed that. I'll to get a python 2 compatible version up tomorrow.

Reply:
Nah, thanks for addressing this issue man.

Reply:
According to [the internet](https://stackoverflow.com/a/624939), it looks like the builtin `callable` should be a solution that works for all 2.X and >=3.2. Is that a safe range for `py4j` at this point?

Reply:
I've verified locally on `2.7`, `3.9`, and `3.10`. I've also verified that in `3.10` an `ImportError` is raised on `from collections import Callable`

Reply:
Yeah it broke python 3.10 support. Or we can even just remove the assert for now.